python/lsst/meas/algorithms/stamps.py

```diff
@@ -144,11 +150,26 @@ def readFitsWithOptions(filename, stamp_factory, options):
                 bbox = Box2I(Point2I(llcX, llcY), Extent2I(width, height))
                 kwargs["bbox"] = bbox
         stamp_parts = {}
+
+        # Determine the dtype from the factory. This allows a Stamp class
+        # to be defined in terms of MaskedImageD or MaskedImageI without
+        # forcing everything to floats.
+        masked_image_cls = None
+        for stamp_field in fields(stamp_factory.__self__):
```

Comment:
I guess you don't want to assume it's named `stamp_im`?

Reply:
Correct. Looking at the `Stamp` code it seems I only require the factory method and that it's a dataclass but it might be confusing if someone calls it something other than `stamp_im`.